OsmAnd/src/net/osmand/plus/GPXUtilities.java

```diff
@@ -989,87 +1009,44 @@ public static GPXFile loadGPXFile(Context ctx, InputStream f) {
 					} else if (parse instanceof GPXExtensions && tag.equals("extensions")) {
 						extensionReadMode = true;
 					} else {
-						if (parse instanceof GPXFile) {
```

Comment:
This is quite a big change and because of the permutation of the lines. Will it be possible to reduce the diff so we can check semantically the changes rather than deep testing.